### ISSUE TITLE
Adding a legend to plots - linked to scalar field

### ISSUE THREAD

**Issue by Nicholas-Mitchell** [daavoo/pyntcloud#235]
I find it great the a scalar field can be named for use a basis for the colour map - I use that all the time.
It would be great though to see a legend next to (or inside) the frame that tells me the range of the values, so I can get a sense for the magnitudes that the colours correspond to.

If this is possible to implement, it could be set up with nice default parameters, that allow a simple toggle argument: `include_legend={True, False}`, and then perhaps an interface like in the Seaborn library, where you can pass an optional dictionary of keyword argument to be passed to the underlying object that draws the legend.

I am not sure there is any alternative option right now - as far as I can tell there is no way to add an indication of the meaning of the colours. Perhaps simply using the neighbouring cell to pretty-print the range of the scalar field used for colouring?

The word "legend" does not appear in the entire [PythreeJS](https://media.readthedocs.org/pdf/pythreejs/latest/pythreejs.pdf) documentation, which makes me think they call it something else...?

**Comment by banesullivan** [comment]
Perhaps this would work with PyVista in #248 

PyVista can easily handle color mapping and rendering color bars and legends in the 3D rendering scene. Perhaps something along the lines of

```py
from pyntcloud import PyntCloud
import pyvista as pv

pc = PyntCloud.from_file("myfile.las")

my_scalars = pc.xyz[:,-1]

dataset = pc.to_pyvista()
dataset['elev'] = my_scalars

dataset.plot(scalars='elev', notebook=False, cmap='viridis')
```

<img width="624" alt="Screen Shot 2019-06-17 at 4 07 16 PM" src="https://user-images.githubusercontent.com/22067021/59639988-00328980-911a-11e9-8c36-5bb6aa726747.png">

**Comment by daavoo** [comment]
Closed in #248
